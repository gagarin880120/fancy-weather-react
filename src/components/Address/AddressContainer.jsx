import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Address from './Address';
import { selectAddress } from '../../redux/selectors';

export function AddressContainer({ address }) {
  return (
    <Address
      address={address}
    />
  );
}

export const mapStateToProps = (state) => ({
  address: selectAddress(state),
});

AddressContainer.propTypes = {
  address: PropTypes.string,

};

AddressContainer.defaultProps = {
  address: '',
};

export default connect(mapStateToProps)(AddressContainer);
