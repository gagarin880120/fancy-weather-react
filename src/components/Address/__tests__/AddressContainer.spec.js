import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { AddressContainer, mapStateToProps } from '../AddressContainer';

describe('AddressContainer', () => {
  describe('component', () => {
    test('should render without crashing', () => {
      const renderer = new ShallowRenderer();
      renderer.render(<AddressContainer />);

      const result = renderer.getRenderOutput();

      expect(result).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {
    test('should return the right value', () => {

      const initialState = {
        address: 'Brest, Belarus',
        countryFlagURL: 'url',
      };

      expect(mapStateToProps(initialState)).toEqual({
        address: 'Brest, Belarus',
        countryFlagURL: 'url',
      });
    });
  })
});
