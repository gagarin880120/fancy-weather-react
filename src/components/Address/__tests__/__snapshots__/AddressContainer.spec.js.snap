// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressContainer component should render without crashing 1`] = `
<Address
  address=""
  countryFlagURL=""
/>
`;
